clients/src/main/java/org/apache/kafka/clients/ClientResponse.java

```diff
@@ -51,7 +51,7 @@ public ClientRequest request() {
         return request;
     }
 
-    public Struct responseBody() {
+    public AbstractResponse responseBody() {
```

Comment:
The casts are annoying, but I thought it was still a step up from blindly trying to parse the struct object as a particular response. The ClassCastException is probably more informative than the error we'd get from parsing an invalid struct and now we can do `instanceof` checks.